planet/src/main/webapp/js/itinerary.js

```diff
@@ -49,19 +63,67 @@ function renderStartingLocation() {
     }
 }
 
+function renderListOptions() {
+    const userId = firebase.auth().currentUser.uid;
+    const eventListRef = database.ref('events/' + userId);
+    const selectListElement = document.getElementById('list-options');
+    // Show the option "current list" regardless of what's stored in the database
+    selectListElement.innerHTML = '<option value="currentList">Current List </option>';
+    eventListRef.once('value', function(eventsSnapshot) {
+        eventsSnapshot.forEach(function(childListSnapshot) {
+            let childKey = childListSnapshot.key;
+            if (childKey !== 'currentList') {
+                let optionElement = document.createElement('option');
+                optionElement.value = childKey;
+                optionElement.innerText = childKey;
+                selectListElement.add(optionElement);
+                // Update the selected value
+                if (childKey === sessionStorage.getItem('listName')) {        
+                    optionElement.selected = true;
+                }
+            }
+        });
+        // Initialize the select tag again since options are dynamically loaded
+        M.FormSelect.init(selectListElement, {});
+    });
+}
+
+function handleListOptionChange() {
+    const listName = document.getElementById ('list-options').value;
+    sessionStorage.setItem('listName', listName);
+    // You can only save currentList as other lists
+    if (listName !== 'currentList') {
+        document.getElementById('save-events-button').style.display = 'none';
+    }else{
+        document.getElementById('save-events-button').style.display = 'inline-block';
+    }
+    renderEvents(listName);
+}
+
 // Add an event to the firebase realtime database
 // Todo: input validation, success/failure callbacks
 async function addEvent() {
     const eventName = document.getElementById('add-event-name').value;
     const eventAddress = document.getElementById('add-event-address').value;
     const eventDuration = document.getElementById('add-event-duration').value;
     const userId = firebase.auth().currentUser.uid; 
+    const listName = document.getElementById('list-options').value;
+
+    // Validate the input fields
+    if ((!eventDuration) || (!eventAddress) || (!eventDuration)) {
+        alert('Please make sure to fill out all fields correctly');
```

Comment:
We should give the user more specific information about what they messed up. Specifically which fields and what is wrong with that field. If you want, you can split this into a validateEventInput() function to keep this code path clean.

Reply:
Okay!